Code/Scripts/fermat_potentials.py

```diff
@@ -0,0 +1,256 @@
+# Imports
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+import numpy as np
+
+import lenstronomy.Util.simulation_util as sim_util
+import lenstronomy.Util.util as util
+from astropy.cosmology import FlatLambdaCDM
+from lenstronomy.Analysis.td_cosmography import TDCosmography
+from lenstronomy.Data.imaging_data import ImageData
+from lenstronomy.LensModel.lens_model import LensModel
+from lenstronomy.LensModel.Solver.lens_equation_solver import LensEquationSolver
+
+# Declaring arrays
+kwargs_model = dict(
```

Comment:
Note that the lens model components are hard-coded here. When switching to include substructure, this will need to be flexible.